### PR TITLE
Added extensions for virtual time testing

### PR DESCRIPTION
**Pull request by mickeelm**
Came across some virtual time-testing scenarios in my own project and I felt that these were missing, wrote them and shared if you are interested in taking them in. 

I'm of course open to discussion and perhaps I'm over-testing them, but I felt it was necessary to show that they actually use the methods that they are intended to bridge.

### CODE REVIEW

**Comment by mickeelm** [comment]
So, this PR is turning four months today. Just a friendly ping...I realize the kotlin-extensions isn't super prio though.

**Comment by simonbasle** [comment]
Damn @mickeelm I'm so sorry... Yeah I haven't looked at the PR list in a long time, looks like I was only subscribed to issues 😢

This PR makes total sense, but I have a comment on the tests.

File: src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt [hunk 1]
```diff
@@ -53,4 +58,49 @@ class StepVerifierExtensionsTests {
                 .verifyError<IllegalStateException>()
     }
 
+    @Test
+    fun `testUsingVirtualTime()`() {
+        { Mono.just("foo").delayElement(Duration.ofDays(1)) }
```
Comment (simonbasle) [inline src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt:63]: maybe, if anything goes wrong, use `Duration.ofMinutes(10)` or something long enough that vts is significant, but if somehow it ends up using realtime the CI won't get stuck for too long?

File: src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt [hunk 2]
```diff
@@ -53,4 +58,49 @@ class StepVerifierExtensionsTests {
...
+            .testUsingVirtualTime()
+            .thenAwait(Duration.ofDays(1))
+            .expectNext("foo")
+            .verifyComplete()
```
Comment (simonbasle) [inline src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt]: alternatively, each of these tests could use the `.expectComplete().verify(Duration.ofSeconds(1))` style here, which would put a timeout on the whole verification

**Review by simonbasle — CHANGES_REQUESTED** [review]
tests can be hardened against time-related bugs, otherwise LGTM

**Comment by mickeelm** [comment]
No worries, I forgot the PR existed even :) 

I'll have a look and make some changes.

File: src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt (hunk 2 above)
Comment (mickeelm) [reply src/test/kotlin/reactor/kotlin/test/StepVerifierExtensionsTests.kt]: I think that's the better approach, so I decided to go with that one. As a bonus, the code is now indented in the same style as Sergei's tests above :)

**Review by simonbasle — APPROVED** [review]
thanks for the PR 👍

**Pull request merged**